### DISCUSSION TITLE
next/script strategy="worker" in App Router

### DISCUSSION THREAD

**Discussion by karlhorky** [vercel/next.js#54431] [Ideas]
Currently, the `next/script` feature to load scripts in web workers via Partytown (`<Script strategy="worker" />`) does not work with the App Router:

> Warning: The `worker` strategy is not yet stable and does not yet work with the [`app`](https://nextjs.org/docs/app/building-your-application/routing/defining-routes) directory. Use with caution.

It would be great to get scripts loaded by a web worker to improve performance, such as is recommended in some places.

### Goals

1. Use [`strategy="worker"`](https://nextjs.org/docs/pages/api-reference/components/script#worker) in App Router
2. Enable loading of third party scripts such as Google Analytics in web workers as [recommended in `third-party-capital`](https://github.com/GoogleChromeLabs/third-party-capital#supported-third-parties) ("offload to a web worker")

### Non-Goals

--

### Background

The feature for the Pages Router was first introduced by @housseindjirdeh in the following PR:

- https://github.com/vercel/next.js/pull/34244

### Proposal

Maybe @housseindjirdeh would be interested in contributing this feature? Or if not, offering some tips on how it could be accomplished?

**Comment by karlhorky** [comment]
Oh I just found the discussion https://github.com/vercel/next.js/discussions/53868, which briefly mentions the `strategy="worker"` mode and some caveats:

> A while back, we rolled out an experimental [worker](https://nextjs.org/docs/pages/building-your-application/optimizing/scripts#offloading-scripts-to-a-web-worker-experimental) strategy to the Script component to allow anyone to move the processing of a third-party script to a web worker. This feature, which uses Partytown, is especially useful when used for tag managers but it can be tricky to set up.

**Comment by AtomSpy** [comment]
Will partytown, which next/script is using under the hood, work in the app router?

**Reply by guybinya** [reply]
Anyone tried it?

**Reply by YashChauhan858** [reply]
It doesn't work

**Comment by danyroza** [comment]
Hi, is there any follow-up on this feature? Alternatively, are there any other ways to load 3rd party scripts in the background when app router is used?
